planner/core/join_explain_test.go

```diff
@@ -0,0 +1,258 @@
+// Copyright 2018 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package core_test
+
+import (
+	"fmt"
+
+	. "github.com/pingcap/check"
+	"github.com/pingcap/tidb/domain"
+	"github.com/pingcap/tidb/kv"
+	"github.com/pingcap/tidb/sessionctx"
+	"github.com/pingcap/tidb/util/mock"
+	"github.com/pingcap/tidb/util/testkit"
+	"github.com/pingcap/tidb/util/testleak"
+)
+
+var _ = Suite(&testSuite{})
+
+type testSuite struct {
+	store kv.Storage
+	dom   *domain.Domain
+	ctx   sessionctx.Context
+}
+
+func (s *testSuite) cleanEnv(c *C) {
+	tk := testkit.NewTestKit(c, s.store)
+	tk.MustExec("use test")
+	r := tk.MustQuery("show tables")
+	for _, tb := range r.Rows() {
+		tableName := tb[0]
+		tk.MustExec(fmt.Sprintf("drop table %v", tableName))
+	}
+}
+
+func (s *testSuite) SetUpSuite(c *C) {
+	var err error
+	testleak.BeforeTest()
+	s.store, s.dom, err = newStoreWithBootstrap()
+	c.Assert(err, IsNil)
+	s.ctx = mock.NewContext()
+}
+
+func (s *testSuite) TearDownSuite(c *C) {
+	s.dom.Close()
+	s.store.Close()
+	testleak.AfterTest(c)()
+}
+
+func (s *testSuite) TestOuterJoinPropConst(c *C) {
```

Comment:
But it doesn't contribue to coverage test as well, since most of the changes is at `expression`, but this test is for `planner/core`. It is better to add unit test for the `PropConstOverOuterJoin`.

Reply:
Make sense, patch updated.